contrail-ui/src/components/main/resource-list-view/ResourceListView.tsx

```diff
@@ -0,0 +1,200 @@
+import TablePagination from "@material-ui/core/TablePagination";
+import TableRow from "@material-ui/core/TableRow";
+import PropTypes from "prop-types";
+import React from "react";
+import styles from "./resourceListStyles";
+import * as types from "./resourceListView.types";
+
+function createData(
+    name: string,
+    resourceId: string,
+    owner: string,
+    dateCreated: string,
+    fileSize: number,
+): types.IData {
+    return { name, resourceId, owner, dateCreated, fileSize };
+}
+
+const rows = [
+    createData("Test.txt", "resId1", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId2", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId3", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId4", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId5", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId6", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId7", "Oliver Song", "Sunday", 16),
+    createData("Test.txt", "resId8", "Oliver Song", "Sunday", 16),
+];
+
+const headRows: types.IHeadRow[] = [
+    { id: "name", numeric: false, label: "Name" },
+    { id: "owner", numeric: false, label: "Owner" },
+    { id: "dateCreated", numeric: false, label: "Date Created" },
+    { id: "fileSize", numeric: true, label: "File Size" },
+];
+
+
+function EnhancedTableHead(props: types.IEnhancedTableProps) {
+    const { onSelectAllClick, numSelected, rowCount } = props;
+
+
+    const renderHeadRows = headRows.map((row) => (
+        <TableCell
+            key={row.id}
+            align={row.numeric ? "right" : "left"}
+            padding="default"
+        >
+            {row.label}
+        </TableCell>
+    ));
+
+    return (
+        <TableHead>
+            <TableRow>
+                <TableCell padding="checkbox">
+                    <Checkbox
+                        indeterminate={numSelected > 0 && numSelected < rowCount}
+                        checked={numSelected === rowCount}
+                        color="default"
+                        onChange={onSelectAllClick}
+                        inputProps={{ "aria-label": "Select all desserts" }}
+                    />
+                </TableCell>
+                {renderHeadRows}
+            </TableRow>
+        </TableHead>
+    );
+}
+
+EnhancedTableHead.propTypes = {
+    numSelected: PropTypes.number.isRequired,
+    onSelectAllClick: PropTypes.func.isRequired,
+    rowCount: PropTypes.number.isRequired,
+};
+
+const useStyles = makeStyles(styles);
+
+export default function EnhancedTable() {
+    const classes = useStyles();
+    const [selected, setSelected] = React.useState<string[]>([]);
+    const [page, setPage] = React.useState(0);
+    const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+    const isSelected = (name: string) => selected.indexOf(name) !== -1;
+
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);
```

Comment:
I think i can somehow minimize this

Reply:
I think this is big brain math

Reply:
Math.min(0, -(rows.length+(page+1) * rowsPerPage));

Reply:
I think instead of minimizing you just made it longer and more complicated

Reply:
m kay